### PR TITLE
[dom-gpu] jupyterlab-1.2.16-CrayGNU-20.06-cuda-batchspawner.eb with Julia[Extensions]

### PR DESCRIPTION
**Pull request by twrobinson**
Jupyterlab 1.2.16 cuda.
This requires JuliaExtensions/1.4.2-CrayGNU-20.06-cuda-10.2 (help wanted, Sam)

### CODE REVIEW

**Comment by twrobinson** [comment]
Awaiting JuliaExtensions/1.4.2-CrayGNU-20.06-cuda-10.2

**Comment by lucamar** [comment]
The pull request #1786 will provide `JupyterLab` with `Julia 1.5.0`: should we close this pull request @omlins?

**Comment by omlins** [comment]
> The pull request #1786 will provide `JupyterLab` with `Julia 1.5.0`: should we close this pull request @omlins?

No, we can keep this open. I will update it to use Julia 1.5.0.

**Comment by lucamar** [comment]
Thanks @omlins: I have updated the name of the recipe and the content to avoid explicit reference to the CUDA version: this should be avoided as well in the dependencies (e.g. `Julia` and `JuliaExtensions`).

File: easybuild/easyconfigs/j/jupyterlab/jupyterlab-1.2.16-CrayGNU-20.06-cuda-10.1-batchspawner.eb [hunk 1]
```diff
@@ -0,0 +1,295 @@
...
+easyblock = 'PythonBundle'
+
+name = 'jupyterlab'
+version = '1.2.16'
+versionsuffix = '-cuda-%(cudashortver)s-batchspawner'
+
+_pyminver = '8'
+
+_jl_maj_ver = '1'
+_jl_min_ver = '4'
```
Comment (omlins) [inline easybuild/easyconfigs/j/jupyterlab/jupyterlab-1.2.16-CrayGNU-20.06-cuda-10.1-batchspawner.eb:12]:
```suggestion
_jl_min_ver = '5'
```

File: easybuild/easyconfigs/j/jupyterlab/jupyterlab-1.2.16-CrayGNU-20.06-cuda-10.1-batchspawner.eb [hunk 2]
```diff
@@ -0,0 +1,295 @@
...
+_jl_rev_ver = '2'
```
Comment (omlins) [inline easybuild/easyconfigs/j/jupyterlab/jupyterlab-1.2.16-CrayGNU-20.06-cuda-10.1-batchspawner.eb:13]:
```suggestion
_jl_rev_ver = '0'
```

**Comment by lucamar** [comment]
Thanks for updating the recipe @omlins: could you please push the recipe `JuliaExtensions-1.5.0-CrayGNU-20.06-cuda.eb` in the branch  `twrobinson-jl1216-cuda`? Otherwise the test will keep failing with this error:
```
ERROR: Missing dependencies: JuliaExtensions/1.5.0-CrayGNU-20.06-cuda
```
See https://jenkins.cscs.ch/blue/organizations/jenkins/TestingEB/detail/TestingEB/3935/pipeline

File: easybuild/easyconfigs/j/jupyterlab/jupyterlab-1.2.16-CrayGNU-20.06-cuda-batchspawner.eb [hunk 3]
```diff
@@ -0,0 +1,295 @@
+# @author: robinson (omlins and hvictor for IJulia)
+
+easyblock = 'PythonBundle'
+
+name = 'jupyterlab'
+version = '1.2.16'
+versionsuffix = '-cuda-batchspawner'
+
+_pyminver = '8'
```
Comment (henrique) [inline easybuild/easyconfigs/j/jupyterlab/jupyterlab-1.2.16-CrayGNU-20.06-cuda-batchspawner.eb:9]:
```suggestion
```
not used anywhere ?

**Comment by omlins** [comment]
I added Julia and JuliaExtensions for PE 20.06 in here.

**Comment by omlins** [comment]
retest this please

File: easybuild/easyconfigs/j/jupyterlab/jupyterlab-1.2.16-CrayGNU-20.06-cuda-batchspawner.eb (hunk 3 above)
Comment (omlins) [reply easybuild/easyconfigs/j/jupyterlab/jupyterlab-1.2.16-CrayGNU-20.06-cuda-batchspawner.eb:9]: Let's wait for Tim to see if he wants to keep / remove it.

**Comment by lucamar** [comment]
The build of `jupyterlab-1.2.16-CrayGNU-20.06-cuda-batchspawner.eb` failed:
see https://jenkins.cscs.ch/blue/organizations/jenkins/TestingEB/detail/TestingEB/3945/pipeline.

The error is the following:
```
Couldn't find file itk_meshtopolydata-0.5.1-cp38-cp38-manylinux1_x86_64.whl anywhere, and downloading it didn't work either
```

**Comment by omlins** [comment]
retest this please

**Comment by twrobinson** [comment]
Hi, please check the 1.1.1 version for a solution for the itk error. Please ask Jean Favre to test it after it is built, as we bumped the version and it is untested.

File: easybuild/easyconfigs/j/jupyterlab/jupyterlab-1.2.16-CrayGNU-20.06-cuda-batchspawner.eb (hunk 3 above)
Comment (twrobinson) [reply easybuild/easyconfigs/j/jupyterlab/jupyterlab-1.2.16-CrayGNU-20.06-cuda-batchspawner.eb:9]: remove, looks stale

**Comment by lucamar** [comment]
I have changed the link to download `itk_meshtopolydata-0.5.1`, since there is no package for Python `3.8` with this version. 
The more recent download for version `0.5.1` is `itk_meshtopolydata-0.5.1-cp37-cp37m-manylinux1_x86_64.whl (531.7 kB)`, unless we upgrade the version of this extension to the latest available `0.6.2`. 
See https://pypi.org/project/itk-meshtopolydata/0.5.1/#files

**Comment by twrobinson** [comment]
Unless you intend to also modify the already merged non-cuda version, I would suggest to instead go with 0.6.0.

**Pull request merged**